csharp/src/Apache.Arrow/ArrowBuffer.Builder.cs

```diff
@@ -120,21 +112,17 @@ public ArrowBuffer Build(MemoryPool pool = default)
                 return new ArrowBuffer(memory);
             }
 
-            private Span<T> EnsureCapacity(int len)
+            private void EnsureCapacity(int n)
             {
-                var targetCapacity = Length + len;
+                var length = Length + n;
```

Comment:
Possible overflow here?

Reply:
Maybe the buffer should internally maintain an "int MaxCapacity" property. Then here we can do:

long newLen = Length + n;
if (newLen > MaxCapacity) 
    throw...

Reply:
Another option is to use the `checked` C# keyword. Then if the addition overflows, it will throw an exception.